DDF-1786 Refactored CswSource refresh to use lambdas and consumers

bdeining opened the pull request:
@kcwire 
@dcruver 
@jaymcnallie 
@jrnorth is hero

review-ninja commented:
<a href="https://app.review.ninja/codice/ddf/pull/455" target="_blank"><img src="https://app.review.ninja/15173563/pull/455/badge" alt="ReviewNinja"/>

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java:
```diff
@@ -388,153 +459,76 @@ public void destroy(int code) {
+            String s = entry.getKey();
+            Consumer c = consumerMap.get(s);
+            if (c != null) {
+                c.accept(configs.get(s));
```
You can use `entry.getValue()` instead of `configs.get(s)`. Also, it would probably be better to use variable names other than `c` and `s`.

dcruver replied:
I would go a step further and instead of using configs.entrySet(), use config.values() since you need all the values and none of the keys..

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java:
```diff
@@ -388,153 +459,76 @@ public void destroy(int code) {
+        // Filter false booleans
+        configs.putAll(configuration.entrySet()
+                .stream()
+                .filter(p -> (p.getValue() instanceof Boolean && !((Boolean) p.getValue())))
```
I'm a bit confused. Do you want to keep the `false` booleans or the `true` booleans? Based on how the "Filter" comment is used in the other two locations, the comment here suggests you want to keep only the `true` booleans.

I suggest breaking these filter operations into their own methods so the code is self-documenting.

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java:
```diff
@@ -388,153 +459,76 @@ public void destroy(int code) {
+        // Filter null integers
+        configs.putAll(configuration.entrySet()
+                .stream()
+                .filter(p -> (p.getValue() instanceof Integer && (p.getValue()) != null))
```
You could just use `p.getValue() instanceof Integer` because that will return false it `p.getValue()` is null.

bdeining commented:
@jrnorth Switched to entry.getValue(), changed some variable names for clarity, broke the filter method into 3 methods, and removed the null check for p.getValue().

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java:
```diff
@@ -304,33 +312,163 @@ private static JAXBContext initJaxbContext() {
+    /**
+     * Consumer function that sets the cswSourceConfiguration ContentType if it is changed.
      */
+    private void setConsumerContentTypeMapping(String currentContentTypeMapping) {
```
It's a bit confusing that this parameter is named `currentContentTypeMapping` because it hasn't been set yet. I like the idea of having a `new` prefix for the parameter like you've used in some of the other setters. At the least, it should be consistent.

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java:
```diff
@@ -801,6 +897,27 @@ public void testRefresh() throws SecurityServiceException {
+                "queryprefix");
+
+        // Assert Defaults
+        assertDefaultCswSourceConfiguration(cswSource.cswSourceConfiguration,
```
Is the purpose of this test to verify the outcome of a refresh with a specific configuration map? If so, I'm a bit confused by the previous two refresh calls since their outcomes aren't being verified. If there's more than one thing being tested here, they should be broken into their own methods.

bdeining replied:
Broke these into their own methods and added assertions.

jrnorth commented on catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java:
```diff
@@ -972,8 +1127,87 @@ private CswSource getCswSource(Csw csw, BundleContext context, String contentMap
-                CswConstants.CSW_NAMESPACE_PREFIX);
+    private void assertDefaultCswSourceConfiguration(CswSourceConfiguration cswSourceConfiguration,
+            CswSourceConfiguration defaultCswSourceConfiguration) {
+        assertThat(cswSourceConfiguration.getUsername(),
```
All these `is(equalTo())` calls could be replaced with `is()`.

bdeining replied:
Done.

bdeining commented on catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java:
```diff
@@ -304,33 +312,163 @@ private static JAXBContext initJaxbContext() {
+    /**
+     * Consumer function that sets the cswSourceConfiguration ContentType if it is changed.
+     */
+    private void setConsumerContentTypeMapping(String newContentTypeMapping) {
```
Changed all consumer helpers to have a `new` prefix in the parameter.

jrnorth commented:
Hero build was successful.
